### PR TITLE
Task 32

### PR DESCRIPTION
**Pull request by elin3t**
DelUserCmd package and test

### CODE REVIEW

File: lib/DelUserCmd.pm [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+use strict;
+use warnings FATAL => 'all';
+use v5.18;
+use parent qw(Command);
+
+use UserService;
+
+sub execute {
+    my $self = shift;
+    my $username = shift;
```
Comment (pauriarte) [inline lib/DelUserCmd.pm]: aqui no recibes por parametro el username. ya la clase padre command tiene una funcion parameters que te devuelve el arreglo de los parametros de tu comando.

**Comment by elin3t** [comment]
@pauriarte Fixed! Now the parameters() method is called to get the username

File: lib/DelUserCmd.pm [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    my $self = shift;
+    my @username = $self->parameters();
+    my $user_service = UserService->new();
+    if ($self->validate($username[0])){
+        return $user_service->delete_user($username[0]);
+    }
+    return 0;
+}
+
+sub validate {
```
Comment (manuellerena) [inline lib/DelUserCmd.pm]: This method is not necessary.

File: lib/DelUserCmd.pm [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+use warnings FATAL => 'all';
+use v5.18;
+use parent qw(Command);
+
+use UserService;
+
+sub execute {
+    my $self = shift;
+    my @username = $self->parameters();
+    my $user_service = UserService->new();
```
Comment (manuellerena) [inline lib/DelUserCmd.pm:12]: Try to inject UserService as a param of the method, with this approach you could mock the method delete_user and isolate the code that you must test.

Comment (elin3t) [inline lib/DelUserCmd.pm:12]: @manuellerena I just remove the validate function :)

**Comment by manuellerena** [comment]
+1

**Comment by earandap** [comment]
+1

**Pull request merged**